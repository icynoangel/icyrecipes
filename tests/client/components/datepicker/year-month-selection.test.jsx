import React from 'react';
import ReactDOM from 'react-dom';
import {mount} from 'enzyme';
import YearMonthSelection from './../../../../src/client/js/components/datepicker/year-month-selection';
import Dropdown from './../../../../src/client/js/components/dropdown/dropdown';
import enzymeToJson from 'enzyme-to-json';

describe('<YearMonthSelection />', function() {
  beforeEach(() => {
    this.onChange = jest.fn();
    this.fromMonth = new Date(2014, 1, 1);
    this.toMonth = new Date(2018, 11, 1);
    this.selectedDay = new Date(2018, 5, 10);
  });

  it('Should render <YearMonthSelection />', () => {
    const wrapper = mount(
      <YearMonthSelection
        onChange={this.onChange}
        fromMonth={this.fromMonth}
        toMonth={this.toMonth}
        selectedDay={this.selectedDay}
      />
    );
    expect(enzymeToJson(wrapper)).toMatchSnapshot();
  });

  it('Should call onChange when changing month', () => {
    const wrapper = mount(
      <YearMonthSelection
        onChange={this.onChange}
        fromMonth={this.fromMonth}
        toMonth={this.toMonth}
        selectedDay={this.selectedDay}
      />
    );

    wrapper
      .find(Dropdown)
      .first()
      .prop('onChange')(10);
    expect(this.onChange).toHaveBeenCalledWith(
      new Date(wrapper.state('year'), 10)
    );
  });

  it('Should not call onChange when changing month and year not set', () => {
    const wrapper = mount(
      <YearMonthSelection
        onChange={this.onChange}
        fromMonth={this.fromMonth}
        toMonth={this.toMonth}
        selectedDay={this.selectedDay}
      />
    );

    wrapper.setState({year: null});
    wrapper
      .find(Dropdown)
      .first()
      .prop('onChange')(10);
    expect(this.onChange).not.toHaveBeenCalled();
  });

  it('Should call onChange when changing year', () => {
    const wrapper = mount(
      <YearMonthSelection
        onChange={this.onChange}
        fromMonth={this.fromMonth}
        toMonth={this.toMonth}
        selectedDay={this.selectedDay}
      />
    );

    wrapper
      .find(Dropdown)
      .last()
      .prop('onChange')(2017);
    expect(this.onChange).toHaveBeenCalledWith(
      new Date(2017, wrapper.state('month'))
    );
  });

  it('Should not call onChange when changing year and month not set', () => {
    const wrapper = mount(
      <YearMonthSelection
        onChange={this.onChange}
        fromMonth={this.fromMonth}
        toMonth={this.toMonth}
        selectedDay={this.selectedDay}
      />
    );

    wrapper.setState({month: null});
    wrapper
      .find(Dropdown)
      .last()
      .prop('onChange')(2017);
    expect(this.onChange).not.toHaveBeenCalled();
  });

  it('Should not update state if selectedDay is the same', () => {
    const wrapper = mount(
      <YearMonthSelection
        onChange={this.onChange}
        fromMonth={this.fromMonth}
        toMonth={this.toMonth}
        selectedDay={this.selectedDay}
      />
    );

    const stateSpy = jest.spyOn(wrapper.instance(), 'setState');
    wrapper.setProps({selectedDay: this.selectedDay});

    expect(stateSpy).not.toHaveBeenCalled();
  });
});
