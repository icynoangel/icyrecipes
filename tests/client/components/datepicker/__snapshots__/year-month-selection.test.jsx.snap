// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<YearMonthSelection /> Should render <YearMonthSelection /> 1`] = `
<YearMonthSelection
  fromMonth={2014-02-01T05:00:00.000Z}
  onChange={[MockFunction]}
  selectedDay={2018-06-10T04:00:00.000Z}
  toMonth={2018-12-01T05:00:00.000Z}
>
  <div
    className="DayPicker-Caption"
  >
    <div
      className="datepicker__month"
    >
      <Dropdown
        items={
          Array [
            Object {
              "label": "January",
              "selected": false,
              "value": 0,
            },
            Object {
              "label": "February",
              "selected": false,
              "value": 1,
            },
            Object {
              "label": "March",
              "selected": false,
              "value": 2,
            },
            Object {
              "label": "April",
              "selected": false,
              "value": 3,
            },
            Object {
              "label": "May",
              "selected": false,
              "value": 4,
            },
            Object {
              "label": "June",
              "selected": true,
              "value": 5,
            },
            Object {
              "label": "July",
              "selected": false,
              "value": 6,
            },
            Object {
              "label": "August",
              "selected": false,
              "value": 7,
            },
            Object {
              "label": "September",
              "selected": false,
              "value": 8,
            },
            Object {
              "label": "October",
              "selected": false,
              "value": 9,
            },
            Object {
              "label": "November",
              "selected": false,
              "value": 10,
            },
            Object {
              "label": "December",
              "selected": false,
              "value": 11,
            },
          ]
        }
        onChange={[Function]}
        placeholder="Month"
        type="tiny"
        width="auto"
      >
        <div
          className="dropdown --tiny --auto"
        >
          <div
            className="dropdown__control"
            onClick={[Function]}
          >
            <div
              className="dropdown__control__select"
            >
              June
            </div>
            <div
              className="dropdown__control__arrow"
            />
          </div>
        </div>
      </Dropdown>
    </div>
    <div
      className="datepicker__year"
    >
      <Dropdown
        items={
          Array [
            Object {
              "label": 2014,
              "selected": false,
              "value": 2014,
            },
            Object {
              "label": 2015,
              "selected": false,
              "value": 2015,
            },
            Object {
              "label": 2016,
              "selected": false,
              "value": 2016,
            },
            Object {
              "label": 2017,
              "selected": false,
              "value": 2017,
            },
            Object {
              "label": 2018,
              "selected": true,
              "value": 2018,
            },
          ]
        }
        onChange={[Function]}
        placeholder="Year"
        type="tiny"
        width="auto"
      >
        <div
          className="dropdown --tiny --auto"
        >
          <div
            className="dropdown__control"
            onClick={[Function]}
          >
            <div
              className="dropdown__control__select"
            >
              2018
            </div>
            <div
              className="dropdown__control__arrow"
            />
          </div>
        </div>
      </Dropdown>
    </div>
  </div>
</YearMonthSelection>
`;
